pkg/cmd/auth/status/status.go

```diff
@@ -97,48 +97,61 @@ func statusRun(opts *StatusOptions) error {
 			statusInfo[hostname] = append(statusInfo[hostname], fmt.Sprintf(x, ys...))
 		}
 
-		if err := shared.HasMinimumScopes(httpClient, hostname, token); err != nil {
-			var missingScopes *shared.MissingScopesError
-			if errors.As(err, &missingScopes) {
-				addMsg("%s %s: the token in %s is %s", cs.Red("X"), hostname, tokenSource, err)
-				if tokenIsWriteable {
-					addMsg("- To request missing scopes, run: %s %s\n",
-						cs.Bold("gh auth refresh -h"),
-						cs.Bold(hostname))
+		scopesHeader, err := shared.GetScopes(httpClient, hostname, token)
+		if err != nil {
+			failed = true
+			addMsg("%s %s: api call failed: %s", cs.Red("X"), hostname, err)
+		} else {
+			if err := shared.HeaderHasMinimumScopes(scopesHeader); err != nil {
+				var missingScopes *shared.MissingScopesError
+				if errors.As(err, &missingScopes) {
+					addMsg("%s %s: the token in %s is %s", cs.Red("X"), hostname, tokenSource, err)
+					if tokenIsWriteable {
+						addMsg("- To request missing scopes, run: %s %s\n",
+							cs.Bold("gh auth refresh -h"),
+							cs.Bold(hostname))
+					}
+				} else {
+					addMsg("%s %s: authentication failed", cs.Red("X"), hostname)
+					addMsg("- The %s token in %s is no longer valid.", cs.Bold(hostname), tokenSource)
+					if tokenIsWriteable {
+						addMsg("- To re-authenticate, run: %s %s",
+							cs.Bold("gh auth login -h"), cs.Bold(hostname))
+						addMsg("- To forget about this host, run: %s %s",
+							cs.Bold("gh auth logout -h"), cs.Bold(hostname))
+					}
 				}
+				failed = true
 			} else {
-				addMsg("%s %s: authentication failed", cs.Red("X"), hostname)
-				addMsg("- The %s token in %s is no longer valid.", cs.Bold(hostname), tokenSource)
-				if tokenIsWriteable {
-					addMsg("- To re-authenticate, run: %s %s",
-						cs.Bold("gh auth login -h"), cs.Bold(hostname))
-					addMsg("- To forget about this host, run: %s %s",
-						cs.Bold("gh auth logout -h"), cs.Bold(hostname))
+				apiClient := api.NewClientFromHTTP(httpClient)
+				username, err := api.CurrentLoginName(apiClient, hostname)
+				if err != nil {
+					addMsg("%s %s: api call failed: %s", cs.Red("X"), hostname, err)
```

Comment:
I believe this should have `failed = true` but don't see a test covering it either way. deferring to the maintainers to make a call here